README.md

```diff
@@ -244,6 +244,156 @@ To request automatic tracing support for a module not on this list, please [file
 
 ## Upgrade guidelines
 
+### 0.17.0 to ???
+
+[PR-1880](https://github.com/open-telemetry/opentelemetry-js/pull/1880) feat(diag-logger): introduce a new global level api.diag for internal diagnostic logging
```

Comment:
I thought we agreed that this whole readme section will become also much smaller and simpler.
From a user perspective we have to mention the basic scenarios,  So I would keep this only:

Defining logger via config option has been removed in favor of global diag logger.

1. Setting logger
```
import { DiagLogLevel, DiagConsoleLogger, diag } from "@opentelemetry/api";
diag.setLogger(new DiagConsoleLogger());
diag.setLogLevel(DiagLogLevel.ERROR);
```
2. Using logger anywhere in a code
```
import { diag } from "@opentelemetry/api";
diag.debug("...");
```
3. Setting logger back to noop
```
import { diag } from "@opentelemetry/api";
diag.setLogger();
```

Reply:
I contemplated that -- which is why I also added the recommended approach to each.
Will remove the "get compiling" steps as it's mostly obvious now I think.